### DISCUSSION TITLE
json vs lock

### DISCUSSION THREAD

**Discussion by adi-wtag** [composer/composer#9777] [Q&A / Support] [answered]
I deleted composer.json file. BUt I have the lock file.
If I run `composer install `it shows an error stating that the composer.json file is missing. Why is it necessary to have a json file for installation ? Doesn't composer install trigger the `composer.lock` file ?

**Comment by stof** [comment, marked as answer]
The lock file contains the information about how dependencies are been resolved. but it does not copy all the information about your root project, which are in the composer.json.

**Reply by adi-wtag** [reply]
Is there any way to retrieve the json file ?

**Reply by stof** [reply]
Well, not from composer. Its job is not to backup your files.

**Reply by adi-wtag** [reply]
Which specific information did you mention in the first comment ? Could you help me with some resources that might clarify the underlying mechanism ? I mean what's actually going on there ? I tried some random blogs but that didn't work.

**Reply by stof** [reply]
well, everything that is in your composer.json:

- autoload config for your root package
- dependency constraints for your root package
- name of your root package (relevant if you have a circular dependency, but for projects rather than libraries, it should not be needed)
- etc...

The composer.json is not meant to be removed once you have updated your dependencies (which generates a lock file). the 2 files are complementary.

**Reply by adi-wtag** [reply]
well, thank you

**Comment by mtarld** [comment]
Hi @adi-wtag,

I have written a simple CLI command to may help you out: https://github.com/mtarld/composer-vendor-scaper
It actually tries to regenerate composer.json from your vendor folder.

**Reply by adi-wtag** [reply]
that's great ! I'll try.
